addons/web/static/src/module_loader.js

```diff
@@ -170,7 +170,46 @@
             }
 
             domReady(() => {
-                // Empty body
+                    const customJSUnloaded = unloaded.some((module) =>
+                    module.includes("user_custom_javascript")
+                );
+
+                if (customJSUnloaded && (document.querySelector(".o_frontend_to_backend_edit_btn") || window.location.href.includes("/web/login"))) {
```

Comment:
`o_frontend_to_backend_edit_btn` is my naive attempt at displaying the error only to users with rights to edit the website. I didn't find a way to obtain the access rights/security groups of the current user.